core/templates/components/state-editor/state-solution-editor/state-solution-editor.component.spec.ts

```diff
@@ -0,0 +1,226 @@
+  }));
+
+  it('should set component properties on initialization', () => {
+    ctrl.$onInit();
+
+    expect($scope.solutionCardIsShown).toBe(true);
+    expect($scope.correctAnswer).toBe(null);
+    expect($scope.inlineSolutionEditorIsActive).toBe(false);
+    expect($scope.SOLUTION_EDITOR_FOCUS_LABEL).toBe(
+      'currentCorrectAnswerEditorHtmlForSolutionEditor');
+    expect($scope.correctAnswerEditorHtml).toEqual(answerEditorHtml);
+  });
+
+  it('should toggle solution card visibility', () => {
+    $scope.solutionCardIsShown = true;
+
+    $scope.toggleSolutionCard();
+
+    expect($scope.solutionCardIsShown).toBe(false);
+
+    $scope.toggleSolutionCard();
+
+    expect($scope.solutionCardIsShown).toBe(true);
+  });
+
+  it('should open delete solution modal when user clicks on delete', () => {
+    spyOn($uibModal, 'open').and.returnValue({
+      result: $q.resolve()
+    });
+    spyOn(StateEditorService, 'deleteCurrentSolutionValidity');
+    spyOn(StateSolutionService, 'saveDisplayedValue');
+
+    $scope.deleteSolution(0, new Event(''));
+    $scope.$apply();
+
+    expect(StateEditorService.deleteCurrentSolutionValidity).toHaveBeenCalled();
+    expect(StateSolutionService.saveDisplayedValue).toHaveBeenCalled();
+  });
+
+  it('should close delete solution modal when user clicks cancel', () => {
+    spyOn($uibModal, 'open').and.returnValue({
+      result: $q.reject()
+    });
+
+    $scope.deleteSolution(0, new Event(''));
+    $scope.$apply();
+
+    expect($uibModal.open).toHaveBeenCalled();
+  });
+
+  it('should get invalid solution tooltip text', () => {
+    spyOn(StateEditorService, 'isInQuestionMode').and.returnValues(true, false);
+
+    // When in question mode.
+    expect($scope.getInvalidSolutionTooltip()).toBe(
+      'This solution doesn\'t correspond to an answer ' +
+      'marked as correct. Verify the rules specified for the ' +
+      'answers or change the solution.'
+    );
+
+    // When not in question mode.
+    expect($scope.getInvalidSolutionTooltip()).toBe(
+      'This solution does not lead to another card. Verify the ' +
+      'responses specified or change the solution.'
+    );
+  });
+
+  it('should check if current solution is valid or not', () => {
+    spyOn(StateEditorService, 'isCurrentSolutionValid').and.returnValue(true);
+    expect($scope.isSolutionValid()).toBe(true);
+  });
+
+  it('should toggle activity of inline solution editor', () => {
+    $scope.inlineSolutionEditorIsActive = true;
+
+    $scope.toggleInlineSolutionEditorIsActive();
+
+    expect($scope.inlineSolutionEditorIsActive).toBe(false);
+
+    $scope.toggleInlineSolutionEditorIsActive();
+
+    expect($scope.inlineSolutionEditorIsActive).toBe(true);
+  });
+
+  it('should get summary of solution', () => {
+    StateSolutionService.savedMemento = solution;
+
+    expect($scope.getSolutionSummary()).toBe(
+      'One solution is "This is a correct answer!".' +
+      ' This is the explanation to the answer.');
+  });
```

Comment:
Insert a pre-check here.

Reply:
Can't have a pre-check as there would be an error if the function is called without initializing `StateSolutionService.savedMemento`